llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -103,6 +87,28 @@ enum CmpPredicate {
   PredSignedLessEqual,
 };
 
+// A binary operator.
+enum BinOp {
+  BinOpAdd,
```

Comment:
Just wanted to point out that these are named this way to avoid name clashes.

In C++ no two enums can have the same variant names.

C++11 introduced something called an `enum class` which lifted this restriction, but disallows treating variants as an integer, which works against us sadly.

After the AOT re-shuffle, I want to audit all the remaining enums and add a prefix to all of their variants.

Reply:
> In C++ no two enums can have the same variant names.

That I did not know!